pkg/sql/sqltelemetry/iam.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2020 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package sqltelemetry
+
+import (
+	"fmt"
+
+	"github.com/cockroachdb/cockroach/pkg/server/telemetry"
+)
+
+// ROLE is used when the syntax used is the ROLE version (ie. CREATE ROLE).
+const ROLE = "role"
+
+// USER is used when the syntax used is the USER version (ie. CREATE USER).
+const USER = "user"
+const iamRoles = "iam.roles"
+
+// IncIAMOption is to be incremented every time a CREATE/ALTER role
+// with an OPTION (ie. NOLOGIN) happens.
+func IncIAMOption(opName string, option string) {
+	telemetry.Inc(telemetry.GetCounter(
+		fmt.Sprintf("%s.%s.%s", iamRoles, opName, option)))
+}
+
+// IncIAMCreate is to be incremented every time a CREATE ROLE happens.
+func IncIAMCreate(typ string) {
```

Comment:
I wonder if there is a way to have a single function to format all of these counter creations. For example, all of these are formatted in the way of "iamroles.operation.type" -- a general function could take in the operation and type to avoid duplicating all of the telemetry.Inc, getCounter and sprintf calls.

Reply:
I can't really think of a nice way to do this since the functions aren't consistent in their parameters.

Reply:
The IAM{Alter,Create,Option,Drop} functions all look similar, IAM{Grant,Revoke} look similar, and IAM{Grant,Revoke}Priviledges.

I also wonder if it is possible to these counters up front with "GetCounterOnce", so that we don't have to lock the telemetry map on every call.

Reply:
Regarding the grouping the functions together - Is there that much value in merging the groups into one function? In this case it'll be difficult to cleanly merge them into one but in my opinion, merging them into 3 separate functions can make it a bit harder to understand. Also the file is fairly small so I think there is some value in having separate functions that are easier to read out at a glance.